synthtool/languages/node.py

```diff
@@ -94,10 +95,48 @@ def get_publish_token(package_name: str):
     return package_name.strip("@").replace("/", "-") + "-npm-token"
 
 
-def postprocess_gapic_library():
+def is_gapic_library():
+    # Pre-requisite #1: google-gax must be a dependency
+    with open("package.json", "r") as package_json:
+        package_json_obj = json.load(package_json)
+    if "dependencies" not in package_json_obj:
+        log.debug("No dependencies in package.json; not a GAPIC library.")
+        return False
+    if "google-gax" not in package_json_obj["dependencies"]:
+        log.debug("Does not depend on google-gax, not a GAPIC library.")
+        return False
+    # Pre-requisite #2: there must be at least one proto list in src/
+    for path in Path("src").rglob("*_proto_list.json"):
+        log.debug(f"Found {path}, it's a GAPIC library.")
+        return True
+    log.debug("No src/**/*_proto_list.json found; not a GAPIC library.")
+    return False
+
+
+def postprocess(install=True, prelint=True, fix=True, protos=True, hide_output=False):
```

Comment:
There are potentially 16 combinations of arguments that could be passed to this function
True, True, True, True,
True, True, True, False,
etc.

Of the 16, how many combinations will actually be used?

Reply:
As of today, we'll have a few `protos=False` (`nodejs-firestore` does weird things with its protos and does not need our post-processing), and I can imagine a library with no samples and thus no `prelint` (`prelint=False`).  I cannot imagine `install=False` or `fix=False` so technically can remove those options if you prefer.

(as for `hide_output=False`, I just don't know yet if we want to be verbose or not in synth scripts)

Do you suggest only leaving those options that we will need right now?

Reply:
I'm open to @SurferJeffAtGoogle's suggestion that we start by exposing the dials and knobs we need immediately, and then introduce additional dials as needed.

Reply:
That sounds like 3 cases.  Could we write 3 functions instead with no branching statements?  And have synth.py call one of the 3 functions?

Reply:
We could - it just sounds like we want to move this ship into different directions :)  I probably don't fully understand what you're going to achieve here. I really want to make all 80 `synth.py` very similar so I would expect all of them just call `node.postprocess()` with no options, and leave options for some one-off cases where the default treatment won't work (e.g. Firestore). So I consider using these boolean options an exception, and having separate functions (e.g. `postprocess_gapic()`, `postprocess_no_samples()`, etc.) will effectively set it these exceptions stone. I'd better add some more intelligence to the post processing and make it understand if this or that action is needed for the given library.

After some thinking - `synthtool` can check if `prelint` is defined in `package.json`, adding a little bit of extra logic to the function, but removing a binary flag. For `protos`, a binary flag will still be needed (because of manual GAPIC snowflakes we have). So, how do you think about this?

```py
def postprocess(compileProtos=True, hide_output=False):
    shell.run(["npm", "install"], hide_output=hide_output)
    if has_prelint():  # checks that scripts.prelint exists in package.json
        shell.run(["npm", "run", "prelint"], hide_output=hide_output)
    shell.run(["npm", "run", "fix"], hide_output=hide_output)
    if protos and is_gapic_library():
        shell.run(["npx", "compileProtos", "src"], hide_output=hide_output)
```

Reply:
No more branching at all.